src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -403,6 +409,8 @@ class spell_warr_heroic_leap : public SpellScriptLoader
         {
             if (WorldLocation* dest = GetHitDest())
                 GetCaster()->CastSpell(dest->GetPositionX(), dest->GetPositionY(), dest->GetPositionZ(), SPELL_WARRIOR_HEROIC_LEAP_JUMP, true);
+            if (GetCaster()->HasAura(SPELL_WARRIOR_BOUNDING_STRIDE))
```

Comment:
this part should be added OnHit no? because caster will get buff after landing
        void HandleStride()
        {
           if (GetCaster()->HasAura(SPELL_WARRIOR_BOUNDING_STRIDE))
               GetCaster()->CastSpell(GetCaster(), SPELL_WARRIOR_BOUNDING_STRIDE_EFFECT, true);
        }
        void Register() override
        {
            OnHit += SpellHitFn(spell_warrior_heroic_leap_SpellScript::HandleStride);
        }

Reply:
no, that won't work here
btw. is the cooldown of taunt also reset after landing?

Reply:
this script do not reset and cooldown i dont even know how to check character spec

Reply:
see https://github.com/TrinityCore/TrinityCore/pull/20839/commits/d1d182dd42a9bf3a65870a269c5e36f329e59e2c

Reply:
Oh Thank You ^_^